autoload/smoothie.vim

```diff
@@ -24,6 +24,14 @@ if !exists('g:smoothie_break_on_reverse')
   let g:smoothie_break_on_reverse = 0
 endif
 
+if !exists('g:smoothie_bell_enabled')
+  ""
+  " Enable beeping when either end of the buffer has been reached, and we
+  " cannot proceed any further
```

Comment:
I think there is no need for a dedicated variable for that. Users already can control whether they want their terminals to beep by manipulating their `belloff` setting. Adding another one seems redundant to me.

Reply:
Oh, I see now that 1cefa9cb3ddddfa56e042de956c2fc1e30e1b34a partially fixes that, by configuring the default in a smart way. I'd go one step further and:
* Remove globally-accessible `g:smoothie_bell_enabled` completely to avoid abstraction leaks;
* Perform the `!(&belloff =~# 'all\|error')` check straight in the timer loop instead of at initialization, so that runtime changes to `belloff` are reflected too.

Reply:
Ok. Will remove it in the next commit.
You finish your review, and I will make a commit containing all the changes you suggest. OK?

Reply:
SGTM 👍